app/Repositories/Project/ProjectRepository.php

```diff
@@ -3,19 +3,97 @@
 namespace App\Repositories\Project;
 
 use App\Models\Project;
+use Storage;
 use App\Repositories\BaseRepository;
 use App\Repositories\Project\ProjectRepositoryInterface;
 use Illuminate\Support\Collection;
+use App\Repositories\Activity\ActivityRepositoryInterface;
+use Illuminate\Http\Request;
 
-class ProjectRepository extends BaseRepository implements ProjectRepositoryInterface
+class ProjectRepository extends BaseRepository implements ProjectRepositoryInterface 
 {
+    
+    private $activityRepository;
     /**
      * ProjectRepository constructor.
      *
      * @param Project $model
      */
-    public function __construct(Project $model)
+    public function __construct(Project $model,ActivityRepositoryInterface $activityRepository) 
     {
+        $this->activityRepository = $activityRepository;
         parent::__construct($model);
     }
+
+    public function clone(Request $request,Project $project) 
+    {
+        $authenticated_user = auth()->user();
+        $token =  $request->bearerToken();
+        $new_image_url = config('app.default_thumb_url');
+        if( Storage::disk('public')->exists( 'projects/'.basename($project->thumb_url) ) && is_file(storage_path("app/public/projects/".basename($project->thumb_url))) ){
+                $ext = pathinfo(basename($project->thumb_url), PATHINFO_EXTENSION);
+                $new_image_name = uniqid().'.'.$ext;
+                ob_start();                
+                \File::copy(storage_path("app/public/projects/".basename($project->thumb_url)) , storage_path("app/public/projects/".$new_image_name) );                
```

Comment:
if/else formatting issue.
And when using operands, please add a 'space' before and after it.
Remove space before comma on line 37, and also remove it after the argument on the same line.